File: Backend/Data/Entities/DisciplineRecord.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Data.Entities
{
    public class DisciplineRecord
    {
        [Required]
        public Discipline Discipline { get; set; }
        [Required]
        public Result BestSingleResult { get; set; }
        [Required]
        public Result BestAverageResult { get; set; }
    }
}

File: Android/app/src/main/java/com/khsm/app/data/entities/DisciplineRecord.java
package com.khsm.app.data.entities;

public class DisciplineRecord {
    public final Discipline discipline;
    public final Result bestSingleResult;
    public final Result bestAverageResult;

    public DisciplineRecord(Discipline discipline, Result bestSingleResult, Result bestAverageResult) {
        this.discipline = discipline;
        this.bestSingleResult = bestSingleResult;
        this.bestAverageResult = bestAverageResult;
    }
}
